src/openrct2/Context.cpp

```diff
@@ -1003,32 +1012,25 @@ namespace OpenRCT2
 
             if (useVariableFrame)
             {
-                RunVariableFrame();
+                RunVariableFrame(deltaTime);
             }
             else
             {
-                RunFixedFrame();
+                RunFixedFrame(deltaTime);
             }
         }
 
-        void RunFixedFrame()
+        void RunFixedFrame(float deltaTime)
         {
-            uint32_t currentTick = platform_get_ticks();
-
-            if (_lastTick == 0)
-            {
-                _lastTick = currentTick;
-            }
-
-            float elapsed = (currentTick - _lastTick) * _timeScale;
-            _lastTick = currentTick;
-            _accumulator = std::min(_accumulator + elapsed, static_cast<float>(GAME_UPDATE_MAX_THRESHOLD));
+            float scaledDeltaTime = deltaTime * _timeScale;
+            _accumulator = std::min(_accumulator + scaledDeltaTime, GAME_UPDATE_MAX_THRESHOLD);
 
             _uiContext->ProcessMessages();
 
             if (_accumulator < GAME_UPDATE_TIME_MS)
             {
-                platform_sleep(GAME_UPDATE_TIME_MS - _accumulator - 1);
```

Comment:
Why is the `-1` gone?

Reply:
Avoiding an underflow, now with the more precise timing this was actually happening